internal/controller/server_controller.go

```diff
@@ -613,3 +645,87 @@ func (r *ServerReconciler) enqueueServerByServerBootConfiguration() handler.Even
 		}
 	})
 }
+
+func (r *ServerReconciler) applyBootOrder(ctx context.Context, log logr.Logger, server *metalv1alpha1.Server) error {
+	if server.Spec.BMCRef == nil && server.Spec.BMC == nil {
+		log.V(1).Info("Server has no BMC connection configured")
+		return nil
+	}
+	bmcClient, err := GetBMCClientForServer(ctx, r.Client, server, r.Insecure)
+	if err != nil {
+		return fmt.Errorf("failed to create BMC client: %w", err)
+	}
+	defer bmcClient.Logout()
+
+	order, err := bmcClient.GetBootOrder(server.Spec.UUID)
+	if err != nil {
+		return fmt.Errorf("failed to create BMC client: %w", err)
+	}
+
+	sort.Slice(server.Spec.BootOrder, func(i, j int) bool {
+		return server.Spec.BootOrder[i].Priority < server.Spec.BootOrder[j].Priority
+	})
+	newOrder := []string{}
+	change := false
+	for i, boot := range server.Spec.BootOrder {
+		newOrder = append(newOrder, boot.Device)
+		if order[i] != boot.Device {
+			change = true
+		}
+	}
+	if change {
+		return bmcClient.SetBootOrder(server.Spec.UUID, newOrder)
+	}
+	return nil
+}
+
+func (r *ServerReconciler) applyBiosSettings(ctx context.Context, log logr.Logger, server *metalv1alpha1.Server) error {
+	serverBase := server.DeepCopy()
+	if server.Spec.BMCRef == nil && server.Spec.BMC == nil {
+		log.V(1).Info("Server has no BMC connection configured")
+		return nil
+	}
+	bmcClient, err := GetBMCClientForServer(ctx, r.Client, server, r.Insecure)
+	if err != nil {
+		return fmt.Errorf("failed to create BMC client: %w", err)
+	}
+	defer bmcClient.Logout()
+
+	version, err := bmcClient.GetBiosVersion(server.Spec.UUID)
+	if err != nil {
+		return fmt.Errorf("failed to create BMC client: %w", err)
+	}
+
+	versionMatch := false
+	diff := map[string]string{}
+	for _, bios := range server.Spec.BIOS {
+		if bios.Version == version {
+			versionMatch = true
+			for key, value := range bios.Settings {
+				if res, ok := server.Status.BIOS.Settings[key]; !ok {
+					if !ok || res != value {
+						diff[key] = value
+					}
+				}
+			}
+			reset, err := bmcClient.SetBiosAttributes(server.Spec.UUID, diff)
+			if err != nil {
+				return err
+			}
+			if reset {
+				server.Status.Conditions = append(server.Status.Conditions, metav1.Condition{
+					Type: "Reboot needed",
+				})
```

Comment:
instead of direct appending condition, I'd be better to use `k8s.io/apimachinery/pkg/api/meta.SetStatusCondition` 

```
SetStatusCondition sets the corresponding condition in conditions to newCondition and returns true if the conditions are changed by this call. conditions must be non-nil.
1. if the condition of the specified type already exists (all fields of the existing condition are updated to newCondition, LastTransitionTime is set to now if the new status differs from the old status)
2. if a condition of the specified type does not exist (LastTransitionTime is set to now() if unset, and newCondition is appended)
```

Reply:
added it